code/game/objects/structures/door_assembly.dm

```diff
@@ -212,17 +212,22 @@
 			src.state = 0
 
 	else if(istype(W, /obj/item/weapon/airlock_electronics) && state == 1)
-		playsound(src.loc, 'sound/items/Screwdriver.ogg', 100, 1)
-		user.visible_message("[user] installs the electronics into the airlock assembly.", "You start to install electronics into the airlock assembly.")
-
-		if(do_after(user, 40))
-			if(!src) return
-			user.drop_item()
-			W.loc = src
-			user << "<span class='notice'>You installed the airlock electronics!</span>"
-			src.state = 2
-			src.name = "Near finished Airlock Assembly"
-			src.electronics = W
+		var/obj/item/weapon/airlock_electronics/EL = W
+		if(!EL.inuse)
+			playsound(src.loc, 'sound/items/Screwdriver.ogg', 100, 1)
+			user.visible_message("[user] installs the electronics into the airlock assembly.", "You start to install electronics into the airlock assembly.")
+			EL.inuse = 1
+			if(do_after(user, 40))
+				EL.inuse = 0
+				if(!src) return
+				user.drop_item()
+				W.loc = src

```

Comment:
For the first point, that code was already there and I just converted it to absolute pathing, but I'll fix that. And for the second one, yup, easy enough to fix.